packages/grafana-data/src/dataframe/DataFrameView.ts

```diff
@@ -1,7 +1,12 @@
-import { DataFrame } from '../types/dataFrame';
+import { DataFrame, Field } from '../types/dataFrame';
 import { DisplayProcessor } from '../types';
 import { FunctionalVector } from '../vector/FunctionalVector';
 
+interface FieldAndValue {
+  field: Field;
+  value: any;
```

Comment:
This is handy, but makes it a bit weird to use:

```ts
view.rows.map((fields, index) => {
  return fields.map(field => field.field); // field.field?
});
```

What if `fields` was a `Field[]`, and you'd still use the index? Might not be reason enough to add it.

```ts
view.rows.map((fields, index) => {
  return fields.map(field => field.values.get(index));
});
```

Or—forgive my TypeScript skills here—could we return a union type that just adds add `value` property to the Field? Might be too much magic though.